plugins/vpp/wireguardplugin/descriptor/peer.go

```diff
@@ -0,0 +1,113 @@
+//  Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package descriptor
+
+import (
+	"fmt"
+	"go.ligato.io/cn-infra/v2/logging"
+	"go.ligato.io/vpp-agent/v3/pkg/models"
+	kvs "go.ligato.io/vpp-agent/v3/plugins/kvscheduler/api"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/wireguardplugin/wgidx"
+
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/wireguardplugin/descriptor/adapter"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/wireguardplugin/vppcalls"
+	wg "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/wireguard"
+)
+
+const (
+	// PeerDescriptorName is the name of the descriptor for VPP wg peer.
+	PeerDescriptorName = "vpp-wg-peer"
+)
+
+// WgPeerDescriptor teaches KVScheduler how to configure VPP wg peer.
+type WgPeerDescriptor struct {
+	log          logging.Logger
+	wgHandler vppcalls.WgVppAPI
+}
+
+// NewWgPeerDescriptor creates a new instance of the wireguard interface descriptor.
+func NewWgPeerDescriptor(wgHandler vppcalls.WgVppAPI, log logging.PluginLogger) *WgPeerDescriptor {
+	return &WgPeerDescriptor{
+		wgHandler:    wgHandler,
+		log:          log.NewLogger("wg-peer-descriptor"),
+	}
+}
+
+// GetDescriptor returns descriptor suitable for registration (via adapter) with
+// the KVScheduler.
+func (d *WgPeerDescriptor) GetDescriptor() *adapter.PeerDescriptor {
+	return &adapter.PeerDescriptor{
+		Name:            PeerDescriptorName,
+		NBKeyPrefix:     wg.ModelPeer.KeyPrefix(),
+		ValueTypeName:   wg.ModelPeer.ProtoName(),
+		KeySelector:     wg.ModelPeer.IsKeyValid,
+		KeyLabel:        wg.ModelPeer.StripKeyPrefix,
+		ValueComparator: d.EquivalentWgPeers,
+		WithMetadata:    true,
+		Create:          d.Create,
+		Delete:          d.Delete,
+		Retrieve:        d.Retrieve,
+	}
```

Comment:
Wireguard peer should have dependencies for VRF table if set to non-0 ([example from code](https://github.com/glazychev-art/vpp-agent/blob/wireguard_plugin/plugins/vpp/l3plugin/descriptor/dhcp_proxy.go#L100-L104))